File: backend/app/schemas/ip_group.py
from datetime import datetime

from pydantic import BaseModel


class IpGroupEntryCreate(BaseModel):
    ip_address: str
    label: str | None = None


class IpGroupEntryResponse(BaseModel):
    id: int
    ip_group_id: int
    ip_address: str
    label: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class IpGroupCreate(BaseModel):
    name: str
    network_id: int
    description: str | None = None
    entries: list[IpGroupEntryCreate] = []


class IpGroupUpdate(BaseModel):
    name: str | None = None
    description: str | None = None


class IpGroupResponse(BaseModel):
    id: int
    name: str
    network_id: int
    network_name: str | None = None
    subnet: str | None = None
    description: str | None = None
    entry_count: int = 0
    entries: list[IpGroupEntryResponse] = []
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/ip-group.ts
export interface IpGroupEntry {
  id: number;
  ip_group_id: number;
  ip_address: string;
  label: string | null;
  created_at: string;
}

export interface IpGroup {
  id: number;
  name: string;
  network_id: number;
  network_name: string | null;
  subnet: string | null;
  description: string | null;
  entry_count: number;
  entries: IpGroupEntry[];
  created_at: string;
  updated_at: string;
}
